src/server/cray/cfs/api/controllers/components.py

```diff
@@ -173,6 +184,58 @@ def patch_components():
     return response, 200
 
 
+def patch_v2_components_dict(data):
+    filters = data.get("filters", {})
```

Comment:
Around this `filters` you are doing alot of manual validation with `filters.get()` is this a good case for some sort of schema or model serialization for the filter data so the field validation is done automatically?. Just a question regarding out API controllers etc.